checks.d/mapreduce.py

```diff
@@ -0,0 +1,499 @@
+                                    and counters['counterGroupName'] is not None:
+
+                                group_name = counters['counterGroupName']
+
+                                if 'counter' in counters \
+                                        and counters['counter'] is not None:
+                                    for counter in counters['counter']:
+
+                                        if 'name' in counter \
+                                                and counter['name'] is not None:
+
+                                            tags = ['cluster_id:' + str(cluster_id),
+                                                    'job_id:' + str(job_id),
+                                                    'group_name:' + str(group_name),
+                                                    'counter_name:' + str(counter['name'])]
+
+                                            self._set_metrics_from_json(tags,
+                                                counter,
+                                                MAPREDUCE_JOB_COUNTER_METRICS)
+
+    def _mapreduce_task_metrics(self, running_jobs, cluster_id):
+        '''
+        Get metrics for each MapReduce task
+        Return a dictionary of {task_id: 'tracking_url'} for each MapReduce task
+        '''
+        running_tasks = {}
+
+        for job_id, tracking_url in running_jobs.iteritems():
+            metrics_json = self._rest_request_to_json(tracking_url,
+                    'tasks',
+                    MAPREDUCE_SERVICE_CHECK)
+
+            if metrics_json:
+                if 'tasks' in metrics_json \
+                        and metrics_json['tasks'] is not None:
+                    if 'task' in metrics_json['tasks'] \
+                            and metrics_json['tasks']['task'] is not None:
+
+                        tasks_json = metrics_json['tasks']['task']
+
+                        for task_json in tasks_json:
+
+                            task_id = task_json['id']
+
+                            running_tasks[task_id] = self._join_url_dir(tracking_url,
+                                'tasks',
+                                task_id)
+
+                            tags = ['cluster_id:' + str(cluster_id),
+                                    'job_id:' + str(job_id),
+                                    'task_id:' + str(task_id)]
+
+                            self._set_metrics_from_json(tags, task_json, MAPREDUCE_TASK_METRICS)
+
+                        self._mapreduce_task_metric_states(tasks_json, job_id, cluster_id)
+
+        return running_tasks
+
+    def _mapreduce_task_metric_states(self, tasks_json, job_id, cluster_id):
+        '''
+        Get metrics for task states and task types
+        '''
+        # Initialize the state and type counts
+        state_counts = {state: 0 for state in MAPREDUCE_TASK_STATE_METRICS}
+        type_counts = {task_type: 0 for task_type in MAPREDUCE_TASK_TYPE_METRICS}
```

Comment:
Just a quick comment: the `default` test fails here on python2.6 because of the dict comprehension that's only in python since 2.7.

You can probably initialize these vars to `defaultdict`s instead (from the [`collections`](https://docs.python.org/2/library/collections.html#collections.defaultdict) module).